ui/js/services/jobfilters.js

```diff
@@ -447,6 +453,20 @@ treeherder.factory('thJobFilters', [
         return locationSearch;
     };
 
+    /**
+     * Same as _stripFieldFilters, but only removes the filter-searchStr
+     * filter from the passed-in locationSearch without
+     * actually setting it in the location bar
+     */
+    var _stripSearchStrFilters = function(locationSearch) {
+        _.forEach(locationSearch, function (val, field) {
+            if (field == "filter-searchStr") {
```

Comment:
s/b ``===`` instead of ``==``.  And better to use ``_withPrefix("searchStr")``  just in case the prefix ever changes.